lib/common/utils.js

```diff
@@ -179,22 +188,22 @@ export const relativeTo = function relativeTo() {
 
 export const flatRoutes = function flatRoutes(router, _path = '', routes = []) {
   router.forEach((r) => {
-    if (!r.path.includes(':') && !r.path.includes('*')) {
-      /* istanbul ignore if */
-      if (r.children) {
-        if (_path === '' && r.path === '/') {
-          routes.push('/')
-        }
-        flatRoutes(r.children, _path + r.path + '/', routes)
-      } else {
-        _path = _path.replace(/^\/+$/, '/')
-        routes.push(
-          (r.path === '' && _path[_path.length - 1] === '/'
-            ? _path.slice(0, -1)
-            : _path) + r.path
-        )
+    if ([':', '*'].some(c => r.path.includes(c))) {
+      return
+    }
+    /* istanbul ignore if */
+    if (r.children) {
+      if (_path === '' && r.path === '/') {
+        routes.push('/')
       }
+      return flatRoutes(r.children, _path + r.path + '/', routes)
     }
+    _path = _path.replace(/^\/+$/, '/')
```

Comment:
If whole path is forward slashes, replace it with a single forward slash?
Probably needs a comment explaining why it's needed, seems a little magic.
`_path` isn't a very descriptive name either.
If you are making  sense of this here stuff, please improve these.

Reply:
In any case, this can be replaced with 
```js
const ONLY_FORWARD_SLASHES_EX = /^\/{2,}$/ //If it's just 1 slash, why bother?
// ...
if (ONLY_FORWARD_SLASHES_EX.test(_path) _path = '/'

Reply:
@qm3ster Agree. Ping the one who wrote it ;)

Reply:
@pi0 Could you please explain?

Reply:
/cc @alexchopin @Atinux